servicetalk-http-api/src/main/java/io/servicetalk/http/api/SingleAddressHttpClientBuilder.java

```diff
@@ -144,4 +144,26 @@
      * @return {@code this}.
      */
     public abstract SingleAddressHttpClientBuilder<U, R> sslConfig(ClientSslConfig sslConfig);
+
+    /**
+     * Disables inference of value to use instead of {@link ClientSslConfig#peerHost()}
+     * from client's address when peer host is not specified.
+     * @return {@code this}
+     */
+    public abstract SingleAddressHttpClientBuilder<U, R> disablePeerHostInference();
+
+    /**
+     * Disables inference of value to use instead of {@link ClientSslConfig#peerPort()}
+     * from client's address when peer port is not specified (equals {@code -1}).
+     * @return {@code this}
+     */
+    public abstract SingleAddressHttpClientBuilder<U, R> disablePeerPortInference();
+
+    /**
+     * Effectively disables <a href="https://datatracker.ietf.org/doc/html/rfc6066#section-3">SNI</a>
+     * and possible SNI hostname inference from client's address if not explicitly specified
+     * via {@link #sslConfig(ClientSslConfig)}.
+     * @return {@code this}
+     */
+    public abstract SingleAddressHttpClientBuilder<U, R> disableSni();
```

Comment:
`disableSni` -> `disableSniInference`? Can you clarify why this method is different than the other new methods?

Reply:
Consider the following names to allow users set/unset the value on the builder:
```
peerHostInference(boolean)
peerPortInference(boolean)
sniHostnameInference(boolean)
```
or
```
inferPeerHost(boolean)
inferPeerPort(boolean)
inferSniHostname(boolean)
```

Reply:
@Scottmitch my reasoning was the following: the inference for SNI is less straightforward than for peerHost – internally, the `X509TrustManagerImpl` does a fallback to peerHost when checking the certificate identity. The internals also [explicitly add](https://github.com/openjdk/jdk8u/blob/4a4236a366eeb961baf157f0938634c1647c447f/jdk/src/share/classes/sun/security/ssl/SSLSocketImpl.java#L1260-L1265) the peerHost to the SNI list.
The existing fallback may seem confusing, hence I decided disabling SNI is the functionality that might be useful. Yet still, my attempt at allowing disabling is not really effective because of the fallback in JDK and Netty. I'll change to the inference related method.